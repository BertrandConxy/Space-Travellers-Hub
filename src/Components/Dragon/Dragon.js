import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getDragonsFromAPI, cancelReservation, setReservation } from '../../Redux/dragon/dragon';
import styles from './dragon.module.css';

const Dragon = () => {
  const allDragons = useSelector((state) => state.dragonReducer);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getDragonsFromAPI());
  }, []);

  const handleReservation = (id) => {
    dispatch(setReservation(id));
  };

  const cancelReserve = (id) => {
    dispatch(cancelReservation(id));
  };

  return (
    <div>
      <ul className={styles['dragon-list']}>
        {allDragons.map((dragon) => (

          <li key={dragon.id} type={dragon.type} className={styles['dragon-item']}>
            <div className={styles['item-img-container']}>
              <img src={dragon.images} alt={dragon.name} />
            </div>
            <div className={styles['item-body']}>
              <h3>{dragon.name}</h3>

              {dragon.reserved && <p>reserved</p>}
              {dragon.reserved ? (
                <button
                  id={dragon.id}
                  className={styles.cancelDragon}
                  type="button"
                  onClick={(e) => {
                    cancelReserve(e.target.id);
                  }}
                >
                  Cancel Reservation
                </button>
              ) : (
                <button
                  id={dragon.id}
                  className={styles.button}
                  type="button"
                  onClick={(e) => {
                    handleReservation(e.target.id);
                  }}
                >
                  Reserve Dragon
                </button>
              )}
            </div>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Dragon;
