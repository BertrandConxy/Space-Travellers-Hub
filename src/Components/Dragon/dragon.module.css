.dragon-list {
  list-style-type: none;
}

.dragon-item {
  display: flex;
  gap: 3rem;
  margin-bottom: 2rem;
}

.dragon-item .item-img-container {
  width: 240px;
  height: 176px;
}

.dragon-item img {
  display: block;
  width: 100%;
  height: 100%;
}

.dragon-item button {
  padding: 8px 10px;
  text-align: center;
  font-size: 14px;
  cursor: pointer;
  border-radius: 4px;
}

.reserve-btn {
  border: 1px solid #1782fc;
  background-color: #1083fd;
  color: #fff;
}

.cancel-btn {
  background: white;
  color: black;
  border: 1px solid rgb(173, 96, 173);
}

.badge {
  color: white;
  background: rgb(60, 180, 180);
  border-radius: 2px;
  width: 50%;
  text-align: center;
}
