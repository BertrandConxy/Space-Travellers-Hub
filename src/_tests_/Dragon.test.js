import renderer from 'react-test-renderer';
import { screen, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import Dragon from '../Components/Dragon/Dragon';

const DragonProvider = () => (
  <Provider store={store}>
    <Dragon />
  </Provider>
);

describe('dragon component test', () => {
  it('reders correctly', () => {
    const dragonComponents = renderer.create(<DragonProvider />).toJSON();
    expect(dragonComponents).toMatchSnapshot();
  });

  it('test from the api', async () => {
    render(<DragonProvider />);
    const allDragons = await screen.findAllByText('Reserve Dragon');
    expect(allDragons).toHaveLength(2);
  });
});
