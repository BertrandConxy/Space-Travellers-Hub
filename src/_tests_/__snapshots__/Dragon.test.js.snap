// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dragon component test reders correctly 1`] = `
<div>
  <ul
    className="dragon-list"
  />
</div>
`;
